OpenRA.Mods.Common/Widgets/Logic/Editor/ActorSelectorLogic.cs

```diff
@@ -72,22 +72,21 @@ public ActorSelectorLogic(Widget widget, World world, WorldRenderer worldRendere
 			selectedOwner = editorLayer.Players.Players.Values.First();
 			Func<PlayerReference, ScrollItemWidget, ScrollItemWidget> setupItem = (option, template) =>
 			{
-				var item = ScrollItemWidget.Setup(template, () => selectedOwner == option, () =>
-				{
-					selectedOwner = option;
-
-					ownersDropDown.Text = selectedOwner.Name;
-					ownersDropDown.TextColor = selectedOwner.Color.RGB;
-
-					InitializeActorPreviews();
-				});
+				var item = ScrollItemWidget.Setup(template, () => selectedOwner == option, () => SelectOwner(option));
 
 				item.Get<LabelWidget>("LABEL").GetText = () => option.Name;
 				item.GetColor = () => option.Color.RGB;
 
 				return item;
 			};
 
+			editorLayer.OnPlayerRemoved = () =>
+			{
+				if (editorLayer.Players.Players.Values.Any(p => p.Name == selectedOwner.Name))
+					return;
+				SelectOwner(editorLayer.Players.Players.Values.FirstOrDefault());
```

Comment:
I think this can get `.First()`. We should always have at least one player anyway. (And this would crash later on as well if we passed `null` here.)